lib/src/test/java/com/knuddels/jtokkit/Cl100kParserTest.java

```diff
@@ -0,0 +1,210 @@
+            String line;
+            while ((line = br.readLine()) != null) {
+                var parts = line.split(";");
+                assert parts.length > 1;
+                var codePoint = Integer.parseInt(parts[0], 16);
+                var name = parts[1];
+                unicodeMap.put(codePoint, name);
+            }
+        } catch (Exception e) {
+            throw new IllegalStateException(e);
+        }
+        return unicodeMap;
+    }
+
+    @Disabled // Takes too long
+    @Test
+    public void testToUtf8BytesOnFetchedUnicodeData() throws Exception {
+        fetchUnicodeData().entrySet().stream().parallel().forEach(e -> {
+            var expected = Character.toString(e.getKey());
+            if (isValidUTF8(expected)) {
+                var dst = new ByteArrayList();
+                addUtf8Bytes(expected, 0, expected.length(), dst);
+
+                assertArrayEquals(expected.getBytes(UTF_8), dst.toArray(), () -> "Expected `" + Arrays.toString(expected.getBytes(UTF_8)) + "` (`" + expected + "`) but was `" + Arrays.toString(dst.toArray()) + "`");
+            } else {
+                System.out.println("Skipping invalid UTF-8: " + e.getValue() + " (" + e.getKey() + ")");
+            }
+        });
+    }
+
+    @Test
+    public void testIsApostophed() {
+        var count = 0;
+        var pattern = compileRegex("^(?:'s|'t|'re|'ve|'m|'ll|'d)$", true);
+
+        System.out.println("isShortContraction");
+        for (var cp1 = MIN_CODE_POINT; cp1 <= MAX_CODE_POINT; cp1++) { // Seems 'ſ is also a contraction...
+            var asString = "'" + Character.toString(cp1);
+            var matchesRegex = pattern.matcher(asString).matches();
+            var actual = Cl100kParser.isShortContraction(cp1);
+            if (matchesRegex) {
+                count++;
+            }
+            assertEquals(matchesRegex, actual, "Mismatch at code point: `" + asString + "` (" + cp1 + ")");
+        }
+
+        if (false) { // Takes too long
+            System.out.println("isLongContraction");
+            for (var cp1 = MIN_CODE_POINT; cp1 <= MAX_CODE_POINT; cp1++) {
+                for (var cp2 = MIN_CODE_POINT; cp2 <= MAX_CODE_POINT; cp2++) {
+                    var asString = "'" + Character.toString(cp1) + Character.toString(cp2);
+                    var matchesRegex = pattern.matcher(asString).matches();
+                    var actual = Cl100kParser.isLongContraction(cp1, cp2);
+                    if (matchesRegex) {
+                        count++;
+                    }
+                    assertEquals(matchesRegex, actual, "Mismatch at code point: `" + asString + "` (" + cp1 + ", " + cp2 + ")");
+                }
+            }
+            System.out.println(count);
+        }
+    }
+
+    @Test
+    public void testIsNumeric() {
+        var count = 0;
+        assertFalse(Cl100kParser.isNumeric(-1));
+        var pattern = compileRegex("^\\p{N}$", true);
```

Comment:
make sure that our `Cl100kParser.isNumeric(cp)` returns the exact same result as `\p{N}` in the original regex by trying it on every single defined character - the tests are surprisingly fast for single characters